lib/dialects/postgres/data-types.js

```diff
@@ -138,6 +194,43 @@ module.exports = BaseTypes => {
     return 'TIMESTAMP WITH TIME ZONE';
   };
 
+  DATE.prototype.validate = function validate(value) {
+    if (value !== Infinity && value !== -Infinity) {
+      return BaseTypes.DATE.prototype.validate.call(this, value);
+    }
+
+    return true;
+  };
+
+  DATE.prototype._stringify = function _stringify(value, options) {
+    if (value === Infinity) {
+      return 'Infinity';
+    } else if (value === -Infinity) {
+      return '-Infinity';
+    }
+
+    return BaseTypes.DATE.prototype._stringify.call(this, value, options);
+  };
+
+  DATE.prototype._sanitize = function _sanitize(value, options) {
+    if ((!options || options && !options.raw) && !(value instanceof Date) && !!value && value !== Infinity && value !== -Infinity) {
+      value = new Date(value);
+    }
+
+    return value;
+  };
+
+  DATE.prototype._isChanged = function _isChanged(value, originalValue) {
+    if (
+      originalValue && !!value &&
+      (value === originalValue || value instanceof Date && originalValue instanceof Date && value.getTime() === originalValue.getTime())
```

Comment:
This logic is again same `Postgres.DATEONLY._isChanged` (which I have suggested to move to `BaseTypes.DATEONLY`)

May be `BaseType.DATE` should have this logic, `BaseType.DATEONLY` should call that and dialect would never need to override it

Reply:
The only reason I extended this into postgres was because it was the only dialect where the `originalValue` and `value` could be equal (`Infinity === Infinity`).  All others would be `Date.getTime() === Date.getTime()`.

However, after some thought I think you are right in moving it into the `BaseType`.